src/test/java/com/fasterxml/jackson/databind/tofix/IPhoneStyleProperty5152Test.java

```diff
@@ -0,0 +1,213 @@
+package com.fasterxml.jackson.databind.tofix;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.testutil.DatabindTestUtil;
+import com.fasterxml.jackson.databind.testutil.failure.JacksonTestFailureExpected;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+// [databind#5152] Support "iPhone" style capitalized properties
+public class IPhoneStyleProperty5152Test
+        extends DatabindTestUtil
+{
+    static class IPhoneBean {
+        private String iPhone;
+
+        public String getIPhone() {
+            return iPhone;
+        }
+
+        public void setIPhone(String value) {
+            iPhone = value;
+        }
+    }
+
+    static class RegularBean {
+        private String phoneNumber;
+
+        public String getPhoneNumber() {
+            return phoneNumber;
+        }
+
+        public void setPhoneNumber(String value) {
+            phoneNumber = value;
+        }
+    }
+
+
+    static class DLogHeaderBean {
+        private String DLogHeader;
+
+        public String getDLogHeader() {
+            return DLogHeader;
+        }
+
+        public void setDLogHeader(String value) {
+            DLogHeader = value;
+        }
+    }
+
+    static class KBSBroadCastingBean {
+        private String KBSBroadCasting;
+
+        public String getKBSBroadCasting() {
+            return KBSBroadCasting;
+        }
+
+        public void setKBSBroadCasting(String value) {
+            KBSBroadCasting = value;
+        }
+    }
+
+    static class PhoneBean {
+        private String Phone;
+
+        public String getPhone() {
+            return Phone;
+        }
+        public void setPhone(String value) {
+            Phone = value;
+        }
+    }
+
+    @JsonPropertyOrder({ "4Roses", "$dollar", "_underscore" })
+    static class NonLetterFirstCharBean {
```

Comment:
No, these are not valid test cases: these will require annotations (like `@JsonProperty`) to be recognized as valid; they will not be auto-detected.

Issue #5152 is specifically about case differences. If there is desire to try to support non-letter-starting property names, let's file different issue.